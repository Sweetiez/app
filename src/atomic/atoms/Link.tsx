import * as React from 'react';
import styled from 'styled-components';

const Content = styled.Text`
  text-decoration: underline;
  margin-right: auto;
  margin-left: auto;
  margin-top: 30px;
`;

interface Props {
  content: string;
  onPress: () => void;
}

const Error: React.FC<Props> = ({content, onPress}) => (
  <Content onPress={onPress}>{content}</Content>
);

export default Error;
