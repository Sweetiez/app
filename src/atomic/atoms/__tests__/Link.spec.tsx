import * as React from 'react';
import renderer from 'react-test-renderer';
import Link from '../Link';

describe('Link', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Link content={'test'} onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
