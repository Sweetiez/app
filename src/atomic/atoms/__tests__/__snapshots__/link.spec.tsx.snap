// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<Text
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 30,
        "textDecorationColor": "black",
        "textDecorationLine": "underline",
        "textDecorationStyle": "solid",
      },
    ]
  }
>
  test
</Text>
`;
